pwndbg/dbg/__init__.py

```diff
@@ -664,6 +664,12 @@ def keys(self) -> List[str]:
         # if there is a better debugger-specific way to do this.
         return [field.name for field in self.fields()]
 
+    def __eq__(self, rhs: object) -> bool:
+        """
+        Returns True if types are the same
+        """
+        raise NotImplementedError()
```

Comment:
Why do we add this?

Reply:
This can be implemented in the base class (here) by using `assert(isinstance(rhs, type(self))), ...`:
<img width="554" alt="image" src="https://github.com/user-attachments/assets/e6a1e495-eab3-4115-bb4d-8723feff4dde">